lib/irb/ext/save-history.rb

```diff
@@ -107,9 +107,9 @@ def save_history
           raise
         end
 
-        if File.exist?(history_file) && @loaded_history_mtime &&
+        if File.exist?(history_file) &&
            File.mtime(history_file) != @loaded_history_mtime
-          history = history[@loaded_history_lines..-1]
+          history = history[(@loaded_history_lines||0)..-1]
```

Comment:
How about 

```rb
history = history[@loaded_history_lines..-1] if @loaded_history_lines
```

because `history[0..-1]` basically means the same as `history`?

And I think it also express the intention clearer. wdyt?

Reply:
Makes sense to me.  I included that change in the commit that fixes the test.